app/views/welcome/dashboard.html.erb

```diff
@@ -5,6 +5,9 @@
         <h2>Announcements</h2>
       </div>
       <div class="card-body">
+        <% policy_scope(Announcement).each do |announcement| %>
+          <%= render announcement if DateTime.now > announcement.start_time && DateTime.now < announcement.end_time %>
```

Comment:
One other thing here - this should probably be a [scope](https://guides.rubyonrails.org/active_record_querying.html#scopes)  instead of filtering them after the fact.